\section{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure Class Reference}
\label{class_p_h_p_unit___framework___test_failure}\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
{\bf \+\_\+\+\_\+construct} ({\bf P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test} \${\bf failed\+Test}, Exception \${\bf thrown\+Exception})
\item 
{\bf to\+String} ()
\item 
{\bf get\+Exception\+As\+String} ()
\item 
{\bf get\+Test\+Name} ()
\item 
{\bf failed\+Test} ()
\item 
{\bf thrown\+Exception} ()
\item 
{\bf exception\+Message} ()
\item 
{\bf is\+Failure} ()
\end{DoxyCompactItemize}
\subsection*{Static Public Member Functions}
\begin{DoxyCompactItemize}
\item 
static {\bf exception\+To\+String} (Exception \${\bf e})
\end{DoxyCompactItemize}
\subsection*{Protected Attributes}
\begin{DoxyCompactItemize}
\item 
{\bf \$failed\+Test}
\item 
{\bf \$thrown\+Exception}
\end{DoxyCompactItemize}


\subsection{Detailed Description}


Definition at line 22 of file Test\+Failure.\+php.



\subsection{Constructor \& Destructor Documentation}
\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!\+\_\+\+\_\+construct@{\+\_\+\+\_\+construct}}
\index{\+\_\+\+\_\+construct@{\+\_\+\+\_\+construct}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{\+\_\+\+\_\+construct}]{\setlength{\rightskip}{0pt plus 5cm}\+\_\+\+\_\+construct (
\begin{DoxyParamCaption}
\item[{{\bf P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test}}]{\$failed\+Test, }
\item[{Exception}]{\$thrown\+Exception}
\end{DoxyParamCaption}
)}\label{class_p_h_p_unit___framework___test_failure_ac4d78cde9700cfcc3601db8a83024f54}
Constructs a Test\+Failure with the given test and exception.


\begin{DoxyParams}[1]{Parameters}
\doxyref{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test}{p.}{interface_p_h_p_unit___framework___test} & {\em \$failed\+Test} & \\
\hline
Exception & {\em \$thrown\+Exception} & \\
\hline
\end{DoxyParams}


Definition at line 45 of file Test\+Failure.\+php.



\subsection{Member Function Documentation}
\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!exception\+Message@{exception\+Message}}
\index{exception\+Message@{exception\+Message}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{exception\+Message}]{\setlength{\rightskip}{0pt plus 5cm}exception\+Message (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)}\label{class_p_h_p_unit___framework___test_failure_acb00ff713fcfbba6ef26c985232d5d28}
Returns the exception\textquotesingle{}s message.

\begin{DoxyReturn}{Returns}
string 
\end{DoxyReturn}


Definition at line 154 of file Test\+Failure.\+php.

\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!exception\+To\+String@{exception\+To\+String}}
\index{exception\+To\+String@{exception\+To\+String}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{exception\+To\+String}]{\setlength{\rightskip}{0pt plus 5cm}static exception\+To\+String (
\begin{DoxyParamCaption}
\item[{Exception}]{\$e}
\end{DoxyParamCaption}
)\hspace{0.3cm}{\ttfamily [static]}}\label{class_p_h_p_unit___framework___test_failure_a987dbaadef3643220df0860dc3b273e3}
Returns a description for an exception.


\begin{DoxyParams}[1]{Parameters}
Exception & {\em \$e} & \\
\hline
\end{DoxyParams}
\begin{DoxyReturn}{Returns}
string 
\end{DoxyReturn}
\begin{DoxySince}{Since}
Method available since Release 3.\+2.\+0 
\end{DoxySince}


Definition at line 90 of file Test\+Failure.\+php.

\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!failed\+Test@{failed\+Test}}
\index{failed\+Test@{failed\+Test}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{failed\+Test}]{\setlength{\rightskip}{0pt plus 5cm}failed\+Test (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)}\label{class_p_h_p_unit___framework___test_failure_afb5353a5bb9ebe3f0b731d176a5f6019}
Returns the failing test.

Note\+: The test object is not set when the test is executed in process isolation.

\begin{DoxySeeAlso}{See also}
\doxyref{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Exception}{p.}{class_p_h_p_unit___framework___exception}
\end{DoxySeeAlso}
\begin{DoxyReturn}{Returns}
P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test$\vert$null 
\end{DoxyReturn}


Definition at line 134 of file Test\+Failure.\+php.

\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!get\+Exception\+As\+String@{get\+Exception\+As\+String}}
\index{get\+Exception\+As\+String@{get\+Exception\+As\+String}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{get\+Exception\+As\+String}]{\setlength{\rightskip}{0pt plus 5cm}get\+Exception\+As\+String (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)}\label{class_p_h_p_unit___framework___test_failure_af5ab125fcd6e125ddf02594c56eadf8b}
Returns a description for the thrown exception.

\begin{DoxyReturn}{Returns}
string 
\end{DoxyReturn}
\begin{DoxySince}{Since}
Method available since Release 3.\+4.\+0 
\end{DoxySince}


Definition at line 78 of file Test\+Failure.\+php.

\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!get\+Test\+Name@{get\+Test\+Name}}
\index{get\+Test\+Name@{get\+Test\+Name}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{get\+Test\+Name}]{\setlength{\rightskip}{0pt plus 5cm}get\+Test\+Name (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)}\label{class_p_h_p_unit___framework___test_failure_af10615e21287d59c3ce1623fa03d79ac}
Returns the name of the failing test (including data set, if any).

\begin{DoxyReturn}{Returns}
string 
\end{DoxyReturn}
\begin{DoxySince}{Since}
Method available since Release 4.\+3.\+0 
\end{DoxySince}


Definition at line 119 of file Test\+Failure.\+php.

\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!is\+Failure@{is\+Failure}}
\index{is\+Failure@{is\+Failure}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{is\+Failure}]{\setlength{\rightskip}{0pt plus 5cm}is\+Failure (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)}\label{class_p_h_p_unit___framework___test_failure_aca7c2d13862869a3538adf2d6a8fc4f5}
Returns true if the thrown exception is of type Assertion\+Failed\+Error.

\begin{DoxyReturn}{Returns}
boolean 
\end{DoxyReturn}


Definition at line 165 of file Test\+Failure.\+php.

\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!thrown\+Exception@{thrown\+Exception}}
\index{thrown\+Exception@{thrown\+Exception}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{thrown\+Exception}]{\setlength{\rightskip}{0pt plus 5cm}thrown\+Exception (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)}\label{class_p_h_p_unit___framework___test_failure_a957f1cb5c23dde4acd265af6a43c87a7}
Gets the thrown exception.

\begin{DoxyReturn}{Returns}
Exception 
\end{DoxyReturn}


Definition at line 144 of file Test\+Failure.\+php.

\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!to\+String@{to\+String}}
\index{to\+String@{to\+String}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{to\+String}]{\setlength{\rightskip}{0pt plus 5cm}to\+String (
\begin{DoxyParamCaption}
{}
\end{DoxyParamCaption}
)}\label{class_p_h_p_unit___framework___test_failure_a5558c5d549f41597377fa1ea8a1cefa3}
Returns a short description of the failure.

\begin{DoxyReturn}{Returns}
string 
\end{DoxyReturn}


Definition at line 63 of file Test\+Failure.\+php.



\subsection{Field Documentation}
\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!\$failed\+Test@{\$failed\+Test}}
\index{\$failed\+Test@{\$failed\+Test}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{\$failed\+Test}]{\setlength{\rightskip}{0pt plus 5cm}\${\bf failed\+Test}\hspace{0.3cm}{\ttfamily [protected]}}\label{class_p_h_p_unit___framework___test_failure_a08708f5d8e67d0b64be0e7b01d950113}


Definition at line 32 of file Test\+Failure.\+php.

\index{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}!\$thrown\+Exception@{\$thrown\+Exception}}
\index{\$thrown\+Exception@{\$thrown\+Exception}!P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure@{P\+H\+P\+Unit\+\_\+\+Framework\+\_\+\+Test\+Failure}}
\subsubsection[{\$thrown\+Exception}]{\setlength{\rightskip}{0pt plus 5cm}\${\bf thrown\+Exception}\hspace{0.3cm}{\ttfamily [protected]}}\label{class_p_h_p_unit___framework___test_failure_ab0eba9a53710b9d8adbdf498eafda753}


Definition at line 37 of file Test\+Failure.\+php.



The documentation for this class was generated from the following file\+:\begin{DoxyCompactItemize}
\item 
vendor/phpunit/phpunit/src/\+Framework/{\bf Test\+Failure.\+php}\end{DoxyCompactItemize}
